File: Objectivity/11.1/samples/cxx/helloWorld/src/HelloObject.cpp
//-----------------------------------------------------------------------------
//  Copyright Objectivity Inc. 2003
//-----------------------------------------------------------------------------
#include <iostream>
#include <ooMap.h>
#include "../ddlFiles/HelloObject.h"

//-----------------------------------------------------------------------------
HelloObject::HelloObject()
   :mHelloMsg(0)
{
}

//-----------------------------------------------------------------------------
const char* 
HelloObject::getHelloMsg() const
{
  return mHelloMsg.head()? mHelloMsg.head() : "";
}

//-----------------------------------------------------------------------------
void 
HelloObject::setHelloMsg(const char* value)
{
  mHelloMsg = value;
}

File: Objectivity/11.1/samples/java/helloWorld/src/HelloObject.java
//-------------------------------------------------------------------
//  Copyright Objectivity Inc. 2012
//-------------------------------------------------------------------
import com.objy.db.app.ooObj;


// Class HelloObject represents the HelloWorld object
public class HelloObject extends ooObj
{
  /** String to contain the message */
  private String mHelloMsg;

  // constructor
  public HelloObject()
  {
  }

  // sets the helloMsg value to the provided value
  void setHelloMsg(String value)
  {
	markModified();
    mHelloMsg = value;
  }

  // returns the current helloMsg value
  String getHelloMsg()
  {
	fetch();
    return mHelloMsg;
  }
}
;
